addons/mail/tests/common.py

```diff
@@ -416,11 +416,25 @@ def assertNoMail(self, recipients, mail_message=None, author=None):
         else:
             raise AssertionError('mail.mail exists for message %s / recipients %s but should not exist' % (mail_message, recipients.ids))
         finally:
-            self.assertNotSentEmail()
+            self.assertNotSentEmail(recipients)
 
-    def assertNotSentEmail(self):
-        """ Check no email was generated during gateway mock. """
-        self.assertEqual(len(self._mails), 0)
+    def assertNotSentEmail(self, recipients=None):
+        """Check no email was generated during gateway mock.
+
+        :param recipients: List of partner for which we will check that no email have been sent
+            If None, we will check that no email at all have been sent
+        """
+        if recipients is None:
+            mails = self._mails
+        else:
+            all_emails = recipients.mapped('email')
```

Comment:
Note that recipients could be a list of emails (a bit hybrid, as sometimes we have partners, sometimes only emails), a bit like what is done in assertEmailSent
```
all_emails = [
    email_to.email if isinstance(email_to, self.env['res.partner'].__class__) else email_to
    for email_to in recipients
]
```